### PR TITLE
[air/train] Remove all references to to_torch/to_tf

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

This should be removed in favor of iter_torch_batches/iter_tf_batches.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by c21** [comment]
Updated all places using `to_torch`, except following places:

4 nightly test:

* https://github.com/ray-project/ray/blob/master/release/nightly_tests/dataset/ray_sgd_runner.py
* https://github.com/ray-project/ray/blob/master/release/nightly_tests/dataset/dataset_shuffle_data_loader.py
* https://github.com/ray-project/ray/blob/master/release/nightly_tests/dataset/ray_sgd_training.py
* https://github.com/ray-project/ray/blob/master/release/nightly_tests/dataset/pipelined_training.py

1 unit test:

* https://github.com/ray-project/ray/blob/master/python/ray/data/tests/test_dataset.py

1 notebook
* https://github.com/ray-project/ray/blob/master/doc/source/ray-air/examples/torch_incremental_learning.ipynb

**Comment by c21** [comment]
Haven't updated `to_tf` yet, I found there's a pattern in codebase to use `to_tf()` together with another API `prepare_dataset_shard()` e.g. in https://github.com/ray-project/ray/blob/master/python/ray/air/examples/tf/tensorflow_linear_dataset_example.py#L57-L58 .

Do we need to change API `prepare_dataset_shard()` as well?

File: doc/source/ray-air/doc_code/tf_starter.py [hunk 1]
```diff
@@ -54,16 +54,17 @@ def train_func(config: dict):
...
-                label_column="y",
-                output_signature=(
-                    tf.TensorSpec(shape=(None, 1), dtype=tf.float32),
-                    tf.TensorSpec(shape=(None), dtype=tf.float32),
-                ),
-                batch_size=batch_size,
-            )
+        tf_dataset = tf.data.Dataset.from_generator(
+            lambda: (
+                (tf.expand_dims(batch["x"], 1), batch["y"])
```
Comment (krfricke) [inline doc/source/ray-air/doc_code/tf_starter.py:59]: We need this dimension expansion in some examples (same for torch), but not in all. Can we get rid of this by adding something to the API?

Also c @c21 here's how to convert to_tf to the new API

**Review by krfricke — COMMENTED** [review]
Continued a bit, but I'm wondering if this API is actually better than the `to_tf` API - to me it seems more complicated. What's the benefit? cc @richardliaw

File: doc/source/ray-air/doc_code/tf_starter.py (hunk 1 above)
Comment (krfricke) [inline doc/source/ray-air/doc_code/tf_starter.py:59]: cc @clarkzinzow @jiaodong for single dimension features, can we prevent auto squeeze (or add a flag)?

**Comment by c21** [comment]
With @krfricke we updated all places using `to_tf`, except following places:

code comment:

* https://github.com/ray-project/ray/blob/master/python/ray/train/tensorflow/train_loop_utils.py
* https://github.com/ray-project/ray/blob/master/python/ray/train/tensorflow/tensorflow_trainer.py

doc:

* https://github.com/ray-project/ray/blob/master/doc/source/train/user_guide.rst
* https://github.com/ray-project/ray/blob/master/python/ray/air/examples/tf/tensorflow_autoencoder_example.py

notebook:

* https://github.com/ray-project/ray/blob/master/doc/source/ray-air/examples/tfx_tabular_train_to_serve.ipynb

**Pull request closed without merging**